addons/binding/org.openhab.binding.dlinkupnpcamera/src/main/java/org/openhab/binding/dlinkupnpcamera/handler/DlinkUpnpCameraHandler.java

```diff
@@ -0,0 +1,327 @@
+                }
+            }
+        } catch (IOException e) {
+            logger.debug("couldn't establish network connection [host '{}']", new Object[] { hostname });
+        }
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        switch (channelUID.getId()) {
+            case IMAGE:
+                refreshImage(command);
+                break;
+            case PAN:
+                try {
+                    sendControlCommand(command, 20, 0);
+                } catch (FileNotFoundException e) {
+                    logger.debug("Command not supported by the camera");
+                }
+                break;
+            case TILT:
+                try {
+                    sendControlCommand(command, 0, 10);
+                } catch (FileNotFoundException e) {
+                    logger.debug("Command not supported by the camera");
+                }
+                break;
+            default:
+                // Nothing is done
+        }
+    }
+
+    private void sendControlCommand(Command command, int x, int y) throws FileNotFoundException {
+        if (command instanceof NextPreviousType || command instanceof PlayPauseType) {
+            switch (command.toString()) {
+                case "NEXT":
+                    sendHttpRequest(buildCommandUrl(hostname, x, y));
+                    break;
+                case "PREVIOUS":
+                    sendHttpRequest(buildCommandUrl(hostname, -x, -y));
+                    break;
+                case "PLAY":
+                    sendHttpRequest(buildPatrolUrl(hostname));
+                    break;
+                case "PAUSE":
+                    sendHttpRequest(buildStopUrl(hostname));
+                    break;
+                default:
+                    // Nothing is done
+            }
+        }
+    }
+
+    private void refreshImage() {
+        URL url = null;
+        try {
+            url = new URL(buildImageUrl(hostname));
+        } catch (MalformedURLException e) {
+            logger.debug("Can't create the URL");
+        }
+        try {
+            updateState(IMAGE, new RawType(readImage(url).toByteArray()));
+        } catch (IOException e) {
+            logger.debug("Can't update the image");
+        }
+    }
+
+    private void refreshImage(Command command) {
+        if (command instanceof RefreshType) {
+            refreshImage();
+        }
+    }
+
+    private ByteArrayOutputStream readImage(URL url) throws IOException {
+        // authentication for network connection
+        Authenticator.setDefault(new Authenticator() {
+            @Override
+            protected PasswordAuthentication getPasswordAuthentication() {
+                return new PasswordAuthentication(getUsername(), getPassword().toCharArray());
+            }
+        });
+
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        InputStream is = null;
+
+        is = url.openStream();
```

Comment:
Using the HttpUrlConnection is surprisingly hard to get right. Especially when considering all the error scenarios, not leaking any resources. For that matter we have the [convention](http://www.eclipse.org/smarthome/documentation/development/bindings/dependencies.html) to use the jetty http client for HTTP communication.

Additionally, I see several potential issues here - the biggest one being that the connection is not closed in case the communication fails.